core/arch/arch.c

```diff
@@ -3383,17 +3383,42 @@ dr_mcontext_to_priv_mcontext(priv_mcontext_t *dst, dr_mcontext_t *src)
                 return false;
         }
         if (TEST(DR_MC_MULTIMEDIA, src->flags)) {
-            /* XXX i#1312: The structure size will double on 64-bit UNIX (possibly
-             * Windows) builds. A corresponding size check will be added with a
-             * future patch.
-             */
 #ifdef X86
             if (src->size > offsetof(dr_mcontext_t, simd)) {
-                if (src->size > offsetof(dr_mcontext_t, simd) +
-                        MCXT_NUM_SIMD_SSE_AVX_SLOTS * YMM_REG_SIZE) {
+                if (MCXT_NUM_SIMD_SLOTS > MCXT_NUM_SIMD_SSE_AVX_SLOTS &&
+                    src->size > offsetof(dr_mcontext_t, simd) +
+                            MCXT_NUM_SIMD_SSE_AVX_SLOTS * ZMM_REG_SIZE) {
+                    if (src->size < offsetof(dr_mcontext_t, simd) + sizeof(dst->simd))
+                        return false;
+                    /* Full copy, UNIX 64-bit. XXX i#1312: We don't support AVX-512
```

Comment:
Confused: isn't the full copy covered by the == case up front?  So why would this be "full copy"?

Reply:
The language is misleading. I meant to say that we're copying the entire sizeof(simd) in these cases. I am re-wording the comment.

Reply:
Re-phrased/clarified.